### ISSUE TITLE
copy.deepcopy of Force objects does not give same type of Force

### ISSUE THREAD

**Issue by jchodera** [pandegroup/openmm#1190]
Somehow `copy.deepcopy` was broken recently, and now gives generic `Force` objects instead of the correct subclasses:
```python
#!/usr/bin/env python                                                                                                                                                                          
from simtk import openmm                                                                                                                                                                       
import copy                                                                                                                                                                                    
force = openmm.NonbondedForce()                                                                                                                                                                
force_copy = copy.deepcopy(force)                                                                                                                                                              
print force                                                                                                                                                                                    
print force_copy
```
gives
```
<simtk.openmm.openmm.NonbondedForce; proxy of <Swig Object of type 'OpenMM::NonbondedForce *' at 0x7f81ada05cf0> >
<simtk.openmm.openmm.Force; proxy of <Swig Object of type 'OpenMM::Force *' at 0x7f80fdc1fc60> >
```
instead of both objects being of type `NonbondedForce`.

**Comment by jchodera** [comment]
The [fast copy implementation](https://github.com/pandegroup/openmm/blob/master/wrappers/python/src/swig_doxygen/swig_lib/python/extend.i#L458) from #1055 may be the culprit.  Here's [the diff](https://github.com/pandegroup/openmm/commit/a707b2b41a60914e2214839a08bbfa86bdb68745) for that commit.

We should also add some python tests that ensure copied `Force` subclasses end up as the same subclass.

**Comment by jchodera** [comment]
Actually, where *are* the Python tests for the Python API?

**Comment by rmcgibbo** [comment]
https://github.com/pandegroup/openmm/tree/master/wrappers/python/tests

**Comment by jchodera** [comment]
Would it be valuable for me to add a Python test in a PR even if I don't
know how to fix the actual bug?  Or should I just wait for @peastman?

**Comment by jchodera** [comment]
Alternatively, I could add a test and revert the change that broke things.

**Comment by peastman** [comment]
Let's not revert it yet.  Hopefully we can find a way to make it work without losing the benefits.

But if we can't find a way to do that, it's probably OK to remove the fast copy implementation for Force and Integrator.  I suspected it's a lot more common to clone Systems than Forces, so that's the most important one.  And Integrators are very cheap to clone.

**Comment by peastman** [comment]
This is fixed by #1219.
